config-store-yaml/src/main/java/io/stargate/config/store/yaml/ConfigStoreYaml.java

```diff
@@ -56,28 +61,41 @@ public ConfigStoreYaml(Path configFilePath, Ticker ticker, MetricRegistry metric
   }
 
   static Map<String, Map<String, Object>> loadConfig(Path configFilePath) throws IOException {
-    return (Map<String, Map<String, Object>>) mapper.readValue(configFilePath.toFile(), Map.class);
+    final File f = configFilePath.toFile();
+    if (!f.exists()) {
+      throw new IOException(
+          String.format(
+              "Can not load YAML config file '%s' (Path '%s'): does not exist", f, configFilePath));
+    }
+    if (!f.canRead()) {
+      throw new IOException(
+          String.format(
+              "Can not load YAML config file '%s' (Path '%s'): not readable", f, configFilePath));
+    }
+    return (Map<String, Map<String, Object>>) mapper.readValue(f, Map.class);
   }
 
   @Override
   public ConfigWithOverrides getConfigForModule(String moduleName)
       throws MissingModuleSettingsException {
     try {
       Map<String, Map<String, Object>> result = configFileCache.get(configFilePath);
-      if (!result.containsKey(moduleName)) {
+      Map<String, Object> config = result.get(moduleName);
+      if (config == null) {
         throw new MissingModuleSettingsException(
             String.format(
-                "The loaded configuration map: %s, does not contain settings from a given module: %s",
-                result, moduleName));
+                "The loaded configuration map (from '%s'): %s, does not contain settings from a given module: %s",
+                configFilePath, result, moduleName));
       }
-      return new ConfigWithOverrides(
-          Collections.unmodifiableMap(result.get(moduleName)), moduleName);
+      logger.info(
```

Comment:
I'm ambivalent about making this info vs debug but I think its ok. this should only happen once in the life of a stargate node right?

Reply:
I definitely hope so. I can easily change it to `debug` if it appears this gets logged too often; that's a valid concern for Integration Tests I think.

Reply:
Alternatively I could move the message to actual loading: How about I'll do that actually...